components/client/src/main/java/com/hotels/styx/client/StyxHttpClientTransaction.java

```diff
@@ -0,0 +1,74 @@
+/*
+  Copyright (C) 2013-2018 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.client;
+
+import com.hotels.styx.api.FullHttpRequest;
+import com.hotels.styx.api.FullHttpResponse;
+import com.hotels.styx.client.netty.connectionpool.NettyConnectionFactory;
+
+import java.util.concurrent.CompletableFuture;
+
+class StyxHttpClientTransaction implements HttpClient.Transaction {
+    private StyxHttpClient.Builder transactionParameters;
```

Comment:
Storing builders violates immutability.
Also these fields are not `final`.

Reply:
I will mark the fields as `final`. Let's keep the discussion about usage of builder as a part of your previous comment. Therefore resolving this finding.